### PR TITLE
Add isRegistered to schemes

### PR DESCRIPTION
**Pull request by tibetsprague**
Also refactor more scheme stuff to scheme/base to reduce duplicate code

And make sure boolean fields can be where queried

### CODE REVIEW

**Comment by tibetsprague** [comment]
@jellegerbrandy @dOrgJelli @orenyodfat do you feel ok about merging this in despite the competition tests failing? im having the hardest time figuring out what is going on and I know @dOrgJelli is already working on fixing them in the client 2.0. we would like this merged in soon

**Comment by dOrgJelli** [comment]
Yes I'm fine with merging this, I can't for the life of me figure out why these tests are failing, they're very spotty :P

Would you be interested in maybe taking them out @jellegerbrandy? Why are we manipulating the apollo client's cache in the competition scheme tests? The 3 tests that are consistently inconsistently failing are:
1. pre-fetching competition.suggestions works
2. pre-fetching competition.suggestions works also without resetting the cache
3. pre-fetching suggestion.votes works

**Comment by orenyodfat** [comment]
@tibetsprague how about doing the "refactor more scheme stuff to scheme/base to reduce duplicate code" in other pr ? so this pr can be cleaner and merged ?

**Comment by tibetsprague** [comment]
@orenyodfat is that really such a big deal? Why would that prevent merging?

**Comment by orenyodfat** [comment]
> @orenyodfat is that really such a big deal? Why would that prevent merging?

it is not..though currently tests does not pass for some reason.

File: test/proposal-competition.spec.ts [hunk 1]
```diff
@@ -451,7 +451,7 @@ describe('Competition Proposal', () => {
     })
   })
 
-  it('Vote state works', async () => {
+  it.skip('Vote state works', async () => {
```
Comment (jellegerbrandy) [inline test/proposal-competition.spec.ts]: hey, why is this test skipped?

**Comment by tibetsprague** [comment]
@jellegerbrandy can i get that approval? and i cant remember if i have permissions to publish a new npm package...

**Pull request merged**